cpp/include/cugraph/algorithms.hpp

```diff
@@ -1536,6 +1537,32 @@ uniform_nbr_sample(raft::handle_t const& handle,
                    std::vector<int> const& h_fan_out,
                    bool with_replacement = true);
 
+/**
+ * @brief Multi-GPU Uniform Neighborhood Sampling.
+ *
+ * @tparam graph_view_t Type of graph view.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Graph View object to generate NBR Sampling on.
+ * @param d_starting_vertices Device span of starting vertex IDs for the NBR Sampling.
+ * @param h_fan_out Host span defining branching out (fan-out) degree per source vertex for each
+ * level
+ * @param with_replacement boolean flag specifying if random sampling is done with replacement
+ * (true); or, without replacement (false); default = true;
+ * @return tuple of tuple of device vectors and counts:
+ * ((vertex_t source_vertex, vertex_t destination_vertex, int rank, edge_t index), rx_counts)
+ */
+template <typename graph_view_t>
```

Comment:
Yeah... and we are sort of mixing

`template <typename graph_view_t>` and using `typename graph_view_t::vertex_type, ...`
and
`template <typename vertext_t, typename edge_t, typename weight_t, bool store_transpoed, bool multi_gpu>` and using `graph_view_t<vertex_t, edge_t, weight_t, store_transposed, multi_gpu>`.

I think we'd better be consistent and any preference in one over the other?

Reply:
No strong preference for me.

There is, I think, and advantage to the `template <typename graph_view_t>` approach in that if we change the implementation of graph_view (adding or removing a template parameter), as long as `typename graph_view_t::vertex_type` is still defined the API works without modification.  I believe Andrei copied this from my Louvain definition which uses this approach.  I implemented Louvain this way so that I could support both the Legacy graph and the graph_t with the same API.

But the syntax is a bit cleaner with your original approach.  I don't think it's likely that we will frequently change the template signature of the API, and we will eventually get rid of the legacy graph class.

I'd be happy to change this back to your original approach, or if we like the `template <typename graph_view_t>` approach better I can add that to the list of things to gradually update in the code.

Reply:
Yeah... I don't have strong preference either but I have strong preference for consistency.

I am also using <template GraphViewType> for primitives but wondering I should better use `graph_view_t<vertex_t, edge_t, weight_t, store_transpoed, multi_gpu>` instead.

I am getting more inclined to the `graph_view_t<vertex_t, edge_t, weight_t, store_transpoed, multi_gpu>` approach as this code does not work for a general graph view type but works only with our graph_view_t (e.g. the implementation depends on multiple member functions only exist in graph_view_t).

And hopefully we can eliminate the legacy code sooner than later; at that point, I slightly prefer `graph_view_t<vertex_t, edge_t, weight_t, store_transpoed, multi_gpu>` even though this will have pretty much very minimal impact on end-user experiences.

Reply:
Sounds good.  I will make those changes in the next push.  I will leave Louvain as it is now.  I plan to create a PR to add Louvain to the C API, I will refactor the Louvain API in that PR.